tests/test_permutation.py

```diff
@@ -69,3 +70,15 @@ def test_permutation_inverse():
     expected = [max(perm_vector)-i for i in perm_vector]
     expected.reverse()
     assert perm.perm_vector == expected
+
+
+@pytest.mark.parametrize("perm_list", [[2, 0, 1], [2, 3, 1, 0]])
+def test_n_mode_permutation_in_2_mode_perms(perm_list):
```

Comment:
You can also add as a test that every components of this circuit is a 2 mode permutation
```
    for _, perm in new_circ:
        assert isinstance(perm, comp.PERM)
        assert perm.m == 2
```

Reply:
Good suggestion :-) Implemented